docs/plugins.md

```diff
@@ -16,51 +16,233 @@ Installing new plugins is very simple, all that is needed is a zip file of plugi
+
+- **EMPTY** | Only available when run outside of a Skelebot project (no skelebot.yaml file present)
+- **CONFIG** | Only available when the Skelebot project has the specified component attributes present in the config
+- **PROJECT** | Available when run inside any Skelebot project
+- **ALWAYS** | Always available no matter what
+
+If the field is not specifed, it will default to the 'CONFIG' activation level.
+
+#### Attributes
+
+Any attributes you need can be added to the plugin class as long as you specify them in the `__init__` function as well. These attributes will be configurable via the corresponding section of the skelebot.yaml components feild.
+
+The name of the section under components will be the camel case version of the plugin's name.
+
+Example:
+```
+components:
+  jupyter:
+    port: 1127
+    folder: .
+```
+
+### Hooks
+
+There are numerous places that a plugin can hook into the Skelebot systems to augment the tool with new features.
+
+#### Scaffold
+
+```
+scaffold(self) # return None
+```
+
+The scaffolder provides a scaffold function that can be implemented in the plugin's Component class in order to add it's own prompts and data into the Skelebot scaffolding process.
+
+The purpose of the scaffold function for a component is to prompt the user for values and assign them to the corresponding attribute in the Component class. These attributes will be automatically recorded int he skelebot.yaml file that is generated.
+
+#### Argument Parsers
+
+```
+addParsers(self,
+    subparsers # the 'job' subparser from the Skelebot ArgumentParser
+) # return subparsers (updated para object)
+```
+
+The argument parser for Skelebot provides a hook called `addParsers` to provide a way for components to supply additional parsing options for the Skelebot CLI. This goes hand-in-hand with the `execute` hook that is shon below.
+
+The `addParsers` function accepts a single parameter, `subparsers`, which is a subparser from the main `argparse.ArgumentParser` object of the Skelebot CLI. Documentation for how to use the parser can be found [here](https://docs.python.org/3/library/argparse.html).
+
+Any parser or argument you add to this parser will be attached to the "job" subparser of the main ArgumentParser. This "job" parser is what will be used by the CLI to determine what gets executed by a given Skelebot command.
+
+This function needs to return the altered subparsers object in order to be properly incorporate the changes into the Skelebot parsing system.
+
+#### Append Dockerignore
+
+```
+appendDockerignore(self) # return string (additional .dockerignore contents)
+```
+
+#### Append Dockerfile
+
+The process of generating the `.dockerignore` file for the project can be augmented using the `appendDockerignore` function. This function takes no parameters, and simply appends the string that is returned from the function to the end of the `.dockerignore` file.
```

Comment:
Pretty minor but I think this paragraph should come before the `Append Dockerfile` section.

Reply:
It absolutely should! Thanks!